ledger/ledgercore/statedelta.go

```diff
@@ -113,7 +115,7 @@ type StateDelta struct {
 	// ModStateProofNextRound represents modification on StateProofNextRound field in the block header. If the block contains
 	// a valid state proof transaction, this field will contain the next round for state proof.
 	// otherwise it will be set to 0.
-	ModStateProofNextRound basics.Round
+	ModStateProofNextRound basics.Round `codec:"StateProofNextRound,omitempty"`
```

Comment:
I think it is better to add a comment that explains why `ModStateProofNextRound ` is the only field that has `codec` tag and why the codec id is different.

Reply:
We're working on some CI automation that would warn when these types diverge from [the ones in the sdk](https://github.com/algorand/go-algorand-sdk/blob/develop/types/statedelta.go#L317).

Hopefully that will serve better than comments like this.

Reply:
are we renaming this field in the sdk, too? it's currently `StateProofNext`.
https://github.com/algorand/go-algorand-sdk/blob/develop/types/statedelta.go#L317

Reply:
@shiqizng thanks, you're right:
```suggestion
	ModStateProofNextRound basics.Round `codec:"StateProofNext,omitempty"`
```